xarray/plot/plot.py

```diff
@@ -48,14 +49,53 @@ def _ensure_plottable(*args):
     axis.
     """
     numpy_types = [np.floating, np.integer, np.timedelta64, np.datetime64]
-    other_types = [datetime]
+    other_types = [datetime, pd.Interval]
 
     for x in args:
         if not (_valid_numpy_subdtype(np.array(x), numpy_types) or
                 _valid_other_type(np.array(x), other_types)):
             raise TypeError('Plotting requires coordinates to be numeric '
                             'or dates of type np.datetime64 or '
-                            'datetime.datetime.')
+                            'datetime.datetime or pd.Interval.')
+
+
+def _interval_to_mid_points(array):
+    """
+    Helper function which returns an array
+    with the Intervals' mid points.
+    """
+
+    return np.asarray(list(map(lambda x: x.mid, array)))
```

Comment:
Consider writing these with list comprehensions, e.g., `np.array([x.mid for x in array])`